pkg/k8s/lease.go

```diff
@@ -0,0 +1,56 @@
+package k8s
+
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+	coordv1 "k8s.io/api/coordination/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func AcquireLease(ctx context.Context, c client.Client, name, namespace, holder string) (*coordv1.Lease, error) {
+	lease := new(coordv1.Lease)
+
+	if err := c.Get(ctx, types.NamespacedName{Namespace: namespace, Name: name}, lease); err != nil {
+		if k8serrors.IsNotFound(err) {
+			lease := &coordv1.Lease{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      name,
+					Namespace: namespace,
+				},
+				Spec: coordv1.LeaseSpec{
+					AcquireTime:    &metav1.MicroTime{Time: time.Now()},
+					HolderIdentity: &holder,
+				},
+			}
+
+			if err := c.Create(ctx, lease); err != nil {
```

Comment:
Given that we have an option for using a fake client for unit testing instead of doing full blown integration tests, would it be easy for us to add them for these 2 exposed functions? Otherwise we can create a ticket for it.